src/utils.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState, useEffect, useCallback } from 'react';
+
+// custom react hook to enable and connect users Marina account.
+function useChecks() {
+  const [isInstalled, setIsInstalled] = useState(false);
+  const [isConnected, setIsConnected] = useState(false);
+
+  const isMarinaInstalled = async () => {
+    if (window.marina) {
+      setIsInstalled(!isInstalled);
+    } else {
+      setIsInstalled(isInstalled);
+    }
+  };
+
+  const checkEnabled = useCallback(async () => {
+    let res = await window.marina.isEnabled();
+    if (res) {
+      setIsConnected(!isConnected);
+    } else {
+        document
+        .getElementById('btn1')
+        .addEventListener('click', async () => await window.marina.enable());
+      document
+        .getElementById('btn2')
+        .addEventListener('click', async () => await window.marina.enable());
+      setIsConnected(!isConnected);
+    }
+  }, []);
+
+  useEffect(() => {
+    isMarinaInstalled();
+    checkEnabled();
```

Comment:
this check should be put in a setInterval, the extension installation & connection can happens later in time after the component is mounted.

An example can be found [here](https://github.com/opendexnetwork/opendex.app/blob/main/src/components/TdexSwapFlow/index.tsx#L76)

Reply:
fixed this